### PR TITLE
Wdb 82/setup docker containers for develop

### PR DESCRIPTION
**Pull request by kibartas**
AMA about the changes. 

Check that the first way to run backend works, at least, the other two are auxiliary.

The Dockerfile is to create an image from the API, the docker-compose.yml is to run both API and DB in two separate containers at the same time.

The Startup.cs change is QOL. No need to `update-database` anymore. (Also helps to keep it simple with the API Docker image)

### CODE REVIEW

File: backend/backend/appsettings.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "ConnectionStrings": {
-    "BackendContext": "Server=.\\SQLEXPRESS;Database=WeDontByte;Trusted_Connection=True;"
+    "BackendContext": "Server=.\\SQLEXPRESS,1433;Database=WeDontByte;User=sa;Password=Password@123;"
```
Comment (AArlauskas) [inline backend/backend/appsettings.json]: This throws exceptions for me. I created a separate user and it works if you do not provide the port. For SQLEXPRESS 1433 is the default port. Is is really necessary to give it as a param in connection string?

File: README.md [hunk 2]
```diff
@@ -2,15 +2,21 @@
...
-3. Open up CMD OR use Package manager console to run command "update-database".
 4. Start the backend project.
 
-# To run the project
-1. Run `yarn start` at at frontend directory
-2. Run x in backend
\ No newline at end of file
+Second way:
+1. Have docker-compose installed
+2. Run `sudo docker-compose up` to launch both DB and backend at the same time (ignore the errors)
```
Comment (AArlauskas) [inline README.md:16]: Ignore the errors sound nice 😆

File: README.md [hunk 3]
```diff
@@ -2,15 +2,21 @@
...
+3. The backend will be running at http://localhost:5000
```
Comment (AArlauskas) [inline README.md]: Can it run on the same port as the IIS (44344)? Then we wouldn't need to change keep changing endpoints in frontend

**Review by AArlauskas — COMMENTED** [review]
Looks good, but need to get the answers and fix the connection to the DB

File: README.md (hunk 2 above)
Comment (kibartas) [reply README.md:16]: Xdd they're really not a big deal

File: README.md (hunk 3 above)
Comment (kibartas) [reply README.md]: Yes, no problem

File: backend/backend/appsettings.json (hunk 1 above)
Comment (kibartas) [reply backend/backend/appsettings.json]: https://social.msdn.microsoft.com/Forums/sqlserver/en-US/e1082cad-9248-44ed-bfcc-c08d7cc08831/login-failed-for-user-sa-microsoft-sql-server-error-18456-severity-14-state-1-when-i-use


Check this link. The port is mandatory for Docker, it seems

Comment (kibartas) [reply backend/backend/appsettings.json]: Reverted the change completely, should work without hiccups in Windows

**Review by AArlauskas — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -2,15 +2,22 @@
...
 
-### Good job, you're done
+## Backend setup
 
-Backend setup
+First way:
 1. You must have SQLEXPRESS installed.
 2. Make sure that your SQLEXPRESS connection string matches the one in appsettings.json.
-3. Open up CMD or use Package manager console to run command "update-database".
 4. Start the backend project.
```
Comment (tmaciulis22) [inline README.md]:
```suggestion
3. Start the backend project.
```

**Review by tmaciulis22 — APPROVED** [review]
Minor change. Did not try to run the backend yet, but it seems that everything will work fine, when I'll try it

**Pull request merged**